src/components/admin/Expense.vue

```diff
@@ -20,27 +20,64 @@
         </li> -->
       </ul>
       <h2>{{expenseMethod}}</h2>
-      <div class="flex-box">
-        <label class="pad-label w100" for="products">
-          <strong>Payee:</strong>
-        </label>
-        <select
-          id="user-dropdown"
-          name="user-dropdown"
-          class="custom-select"
-          v-model="transaction.payee"
-        >
-          <option :value="-1">SELF</option>
-          <option disabled>--- username - company ---</option>
-          <option class="batches-op" v-for="item in users" v-bind:key="item.id" v-bind:value="item.id">
-            <span>{{item.username}} - {{item.company.company_name}}</span>
-          </option>
-          <option disabled>--- vendor - phone - company ---</option>
-          <option class="batches-op" v-for="item in vendors" v-bind:key="item.id" v-bind:value="item.id">
-            <span>{{item.first_name}} - {{item.username}} - {{item.company.company_name}}</span>
-          </option>
-        </select>
-      </div>
+      <template v-if="expenseMethod === 'Credit'">
+        <div class="flex-box">
+          <label class="pad-label w100" for="products">
+            <strong>Payor:</strong>
+          </label>
+          <select
+            id="user-dropdown"
+            name="user-dropdown"
+            class="custom-select"
+            v-model="transaction.payor"
+          >
+            <option :value="-1">SELF</option>
+            <template v-if="userdata.company.company_type == 'RETIAL'">
+              <option disabled>--- Regular Customers---</option>
+              <option class="batches-op" v-for="item in users.filter(usr => usr.user_type == 'REGULAR_CUSTOMER')"
+                v-bind:key="item.id" v-bind:value="item.id">
+                <span>{{item.username}} - {{item.company.company_name}}</span>
+              </option>
+            </template>
+            <template v-if="userdata.company.company_type == 'STORE'">
+              <option disabled>--- username - company ---</option>
+              <option class="batches-op" v-for="item in users.filter(usr => usr.company.company_type == 'STORE')"
```

Comment:
this code is repeating for retail and store. 
We can do this inside some getter. Like instead of filtering here we can filter inside the getter based on the userdata.company.company_type